### PR TITLE
[1747] Rename course enrichment qualifications field

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
Our course model has `qualifications` as well as `qualifications` from the enrichment model which is confusing and caused us to add some "workarounds"

### Changes proposed in this pull request
Rename `qualifications` to `required_qualifications` in the course enrichment model

### Guidance to review
- Run the migration then check a course enrichment 

### Checklist
- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by dankmitchell** [comment]
[Fix inconsistent qualifications/required_qualifications field names](https://trello.com/c/BnKmktuz/1747-fix-inconsistent-qualifications-requiredqualifications-field-names)

File: app/models/course_enrichment.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ class CourseEnrichment < ApplicationRecord
                  interview_process: [:string, store_key: 'InterviewProcess'],
                  other_requirements: [:string, store_key: 'OtherRequirements'],
                  personal_qualities: [:string, store_key: 'PersonalQualities'],
-                 qualifications: [:string, store_key: 'Qualifications'],
+                 required_qualifications: [:string, store_key: 'RequiredQualifications'],
```
Comment (defong) [inline app/models/course_enrichment.rb]: https://github.com/DFE-Digital/manage-courses-api/blob/7419131f3fb92e41c8f131281afaba96e2abb1ad/src/ManageCourses.Api/Model/CourseEnrichmentModel.cs#L16

are you sure?

**Comment by defong** [comment]
This PR calls for babysitting @misaka

**Comment by tvararu** [comment]
What @defong suggested makes sense; we could do this as a more increment change which is easier to deploy and less error prone

**Comment by misaka** [comment]
We could also just focus on moving the background publishing job to Rails and then change it then.

**Comment by misaka** [comment]
@dankmitchell Maybe this PR should just add a comment to not try changing the name of this attribute until kill C#. :)

File: db/migrate/20190722124603_rename_course_enrichment_qualifications.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class RenameCourseEnrichmentQualifications < ActiveRecord::Migration[5.2]
+  def up
+    execute "UPDATE   course_enrichment
+                SET   json_data = Replace(json_data :: text, '\"Qualifications\":',
+                              '\"RequiredQualifications\":') ::
+                              jsonb
+              WHERE   json_data :: text LIKE '%\"Qualifications\":%';"
+  end
```
Comment (misaka) [inline db/migrate/20190722124603_rename_course_enrichment_qualifications.rb]: I'm a little confused by this. Isn't the model now expecting `required_qualifications` to be the JSON field `Qualifications`? And isn't this replacing `"Qualifications"` with `"RequiredQualifications"`? Which would then break `required_qualifications`?

Comment (defong) [reply db/migrate/20190722124603_rename_course_enrichment_qualifications.rb]: I don't think the migration file is needed. Dan probably forgot about it.

ie, Data should be stored as `qualification` and then get hydrated as `required_qualification`

Comment (misaka) [reply db/migrate/20190722124603_rename_course_enrichment_qualifications.rb]: Ok, so this needs to be removed, I think.

**Comment by misaka** [comment]
:shipit:

**Pull request merged**